src/Microsoft.AspNet.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http
+{
+    public class DateHeaderValueManager : IDisposable
+    {
+        private readonly TimeSpan _timeWithoutRequestsUntilIdle = TimeSpan.FromSeconds(10);
+        private readonly TimeSpan _timerInterval = TimeSpan.FromSeconds(1);
+        private readonly uint _timerTicksWithoutRequestsUntilIdle;
+
+        private volatile string _dateValue;
+        private bool _isDisposed = false;
+        private bool _hadRequestsSinceLastTimerTick = false;
+        private Timer _dateValueTimer;
+        private object _timerLocker = new object();
+        private int _timerTicksSinceLastRequest;
+
+        public DateHeaderValueManager()
+        {
+            _timerTicksWithoutRequestsUntilIdle = (uint)(_timeWithoutRequestsUntilIdle.TotalMilliseconds / _timerInterval.TotalMilliseconds);
+        }
+
+        public string GetDateHeaderValue()
+        {
+            PumpTimer();
+
+            return _dateValue ?? DateTime.UtcNow.ToString("r");
+        }
+        
+        public void Dispose()
+        {
+            lock (_timerLocker)
+            {
+                if (_dateValueTimer != null)
+                {
+                    DisposeTimer();
+                    _isDisposed = true;
+                }
+            }
+        }
+
+        private void PumpTimer()
+        {
+            _hadRequestsSinceLastTimerTick = true;
+
+            if (!_isDisposed && _dateValueTimer == null)
+            {
+                lock (_timerLocker)
+                {
+                    if (!_isDisposed && _dateValueTimer == null)
+                    {
+                        // Immediately assign the date value and start the timer again
+                        _dateValue = DateTime.UtcNow.ToString("r");
+                        _dateValueTimer = new Timer(UpdateDateValue, state: null, dueTime: _timerInterval, period: _timerInterval);
```

Comment:
Why not create the timer in the constructor and just do `Change()` here? Then, `_dateValueTimer != null` is an invariant and wouldn't need to be checked all the time.

Also: Why not use `dueTime: TimeSpan.Zero` to run the handler immediately, which sets the `_dateValue` immediately.